cnxdb/migrations/20170911200000_update_index_cnxml.py

```diff
@@ -0,0 +1,80 @@
+# -*- coding: utf-8 -*-
+
+
+def up(cursor):
+    # Select all the modules that have both index.cnxml and
+    # index_auto_generated.cnxml
+    cursor.execute("""\
+SELECT module_ident, array_agg(fileid),
+       array_agg(substring(
+           encode(file, 'escape') from 'cnxml-version=.....'))
+    FROM module_files NATURAL JOIN files
+    WHERE filename IN ('index.cnxml', 'index_auto_generated.cnxml')
+    GROUP BY module_ident HAVING count(*) > 1""")
+    rows = cursor.fetchall()
+
+    if not rows:
+        # nothing to do
+        return
+
+    # Create table for rollback
+    module_idents = [r[0] for r in rows]
+    cursor.execute("""\
+CREATE TABLE datamigrations.index_auto_generated AS
+    SELECT * FROM module_files
+        WHERE module_ident IN %s
+          AND filename IN ('index.cnxml', 'index_auto_generated.cnxml')
+""", (tuple(module_idents),))
+
+    for row in rows:
+        module_ident, fileids, cnxml_versions = row
+        if cnxml_versions == [None, None]:
+            # Skip files that have no cnxml version
+            continue
+        elif cnxml_versions[0] is None:
+            keep_fileid = fileids[1]
+        elif cnxml_versions[1] is None:
+            keep_fileid = fileids[0]
+        else:
+            # Choose the higher fileid if the cnxml versions are the same
+            if cmp(*cnxml_versions) == 0:
+                keep_fileid = max(fileids)
+            elif cmp(*cnxml_versions) == 1:
+                keep_fileid = fileids[0]
+            else:
+                keep_fileid = fileids[1]
```

Comment:
I've just surveyed my local db, and I think we just came up with a complicated way to say "always pick index_auto_generated.cnxml". :-) But that looks to be ok, for the ones I spot-checked. The only diff seems to be character encoding